File: go/wam/4870.go
const WamMetaVerifiedInteraction = 4870 // channel: private

type MetaVerifiedInteractionEvent struct {
	BusinessOwnerJid string `wam:"business_owner_jid"` // field 1
	BusinessOwnerPlatform BUSINESSOWNERPLATFORM `wam:"business_owner_platform"` // field 2
	MetaVerifiedInteractionAction METAVERIFIEDINTERACTIONACTION `wam:"meta_verified_interaction_action"` // field 3
	MetaVerifiedInteractionAssetType METAVERIFIEDINTERACTIONASSETTYPE `wam:"meta_verified_interaction_asset_type"` // field 4
	MetaVerifiedInteractionReferral METAVERIFIEDINTERACTIONREFERRAL `wam:"meta_verified_interaction_referral"` // field 5
	MetaVerifiedInteractionSurface METAVERIFIEDINTERACTIONSURFACE `wam:"meta_verified_interaction_surface"` // field 6
	IsMetaVerifiedSubscribed bool `wam:"is_meta_verified_subscribed"` // field 7
	IsSelfView bool `wam:"is_self_view"` // field 8
}

File: ts/wam/4870.ts
export const WamMetaVerifiedInteraction = 4870 as const

export interface MetaVerifiedInteractionEvent {
  /** field 1, wire `business_owner_jid` */
  businessOwnerJid?: string
  /** field 2, wire `business_owner_platform` */
  businessOwnerPlatform?: BUSINESSOWNERPLATFORM
  /** field 3, wire `meta_verified_interaction_action` */
  metaVerifiedInteractionAction?: METAVERIFIEDINTERACTIONACTION
  /** field 4, wire `meta_verified_interaction_asset_type` */
  metaVerifiedInteractionAssetType?: METAVERIFIEDINTERACTIONASSETTYPE
  /** field 5, wire `meta_verified_interaction_referral` */
  metaVerifiedInteractionReferral?: METAVERIFIEDINTERACTIONREFERRAL
  /** field 6, wire `meta_verified_interaction_surface` */
  metaVerifiedInteractionSurface?: METAVERIFIEDINTERACTIONSURFACE
  /** field 7, wire `is_meta_verified_subscribed` */
  isMetaVerifiedSubscribed?: boolean
  /** field 8, wire `is_self_view` */
  isSelfView?: boolean
}